### PR TITLE
DOC: Updating str_slice docstring

### PR DESCRIPTION
**Pull request by JesperDramsch**
Added Examples to docstring.
Added documentation to each part of docstring

### CODE REVIEW

**Review by gfyoung — APPROVED** [review]
Nice changes!

cc @jreback @datapythonista

File: pandas/core/strings.py [hunk 1]
```diff
@@ -1385,17 +1385,40 @@ def str_rsplit(arr, pat=None, n=None):
...
+    0    nda
+    1      x
+
+    >>> s.str.slice(stop=2)
+    0    pa
+    1    fo
+
+    >>> s.str.slice(step=2)
+    0    pna
+    1     fx
```
Comment (datapythonista) [inline pandas/core/strings.py]: May be it's worth showing that `s.str[2:-2]` (or whatever slice) is equivalent to `s.str.slice(start=2, stop=-2)`.

**Review by datapythonista — COMMENTED** [review]
Nice changes, added some comments about formatting and ideas to make the docstring even better.

**Comment by jorisvandenbossche** [comment]
@JesperDramsch do you have time to update this PR based on the comments?

**Comment by JesperDramsch** [comment]
> @JesperDramsch do you have time to update this PR based on the comments?

Totally missed this one. I'll get to it asap

**Comment by JesperDramsch** [comment]
Here we go. Ran validation and updated everything according to the comments and linter. Thank you for the comments @datapythonista and thanks for the patience @jorisvandenbossche.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks for another great docstring @JesperDramsch 

The build is broken now, but will merge once it's fixed and the checks are green.

**Comment by datapythonista** [comment]
@JesperDramsch can you rebase, so the CI runs again, and we can merge it?

**Comment by JesperDramsch** [comment]
> @JesperDramsch can you rebase, so the CI runs again, and we can merge it?

I'm not really sure what that means. How do I rebase?
Do you have a link?

**Comment by datapythonista** [comment]
Sorry, we use rebase as short for updating your PR with the latest master.

Just run in your branch: `git fetch upstream && git merge upstream/master && git push`.

This is usually required when the branch of the PR has conflicts with master. In this case github would manage the merge itself, as there are no conflicts. But as the checks of the CI failed (for reasons unrelated to your PR), we just want them to run again. And as there are many systems, and I don't thing I even have permissions to all them, the easiest is that you update your branch, and then all them are automatically rerun.

With some luck we'll make things easier soon. :)

**Comment by JesperDramsch** [comment]
Alright. I can get to that tomorrow easily.

File: pandas/core/strings.py [hunk 2]
```diff
@@ -1385,17 +1385,68 @@ def str_rsplit(arr, pat=None, n=None):
...
+    dtype: object
+
+    >>> s.str.slice(start=0, stop=5, step=3)
+    0    kl
+    1     f
+    2    cm
+    dtype: object
+
+    Equivalent behaviour to:
+    >>> s.str[0:5:3]
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: An empty line is needed between the explanation and the code line starting with `>>>`

**Comment by datapythonista** [comment]
@JesperDramsch seems like there is a formatting issue `pandas/core/strings.py:1497:1: W293 blank line contains whitespace`, that's why travis is failing.

Can you fix it please. Happy to merge it afterwards.

**Comment by JesperDramsch** [comment]
Aw Yiss.

**Comment by datapythonista** [comment]
Thanks for another great docstring @JesperDramsch

**Pull request merged**